Fix double encoding

sbishel opened the pull request:
#### Summary
Fixes issue where links are encoded even thought they may already be encoded. Fix is simply decode prior to encoding. 

#### Ticket Link

  Fixes https://github.com/mattermost/focalboard/issues/3496

Pinjasaur reviewed (approved):
overall LGTM @sbishel, could you update the test or add a new test to assert the `title` too?

sbishel commented:
> overall LGTM @sbishel, could you update the test or add a new test to assert the `title` too?

Nope, it looks like that line can actually be removed. We have no way of setting it with this function. My test data just happens to use "title" as a query parameter, but they are unrelated.

Pinjasaur commented:
@sbishel, sorry didn't see the collision of the `title` query parameter and the `title` HTML attribute. I was talking about the attribute here. I _think_ it can be set using this syntax: https://www.markdownguide.org/basic-syntax/#adding-titles

sbishel commented:
@Pinjasaur - Nice catch. By encoding the title we end up with encoded values in the tooltip. Actually appears that "title" does not need to be encoded/decoded at all, fixed and added additional tests.

sbishel commented:
@Pinjasaur - I know you approved earlier, but please review again for latest changes.

Pinjasaur commented on webapp/src/utils.test.ts:
```diff
@@ -58,6 +58,16 @@ describe('utils', () => {
             window.openInNewBrowser = null
         })
+
+        test('should encode title', () => {
```
```suggestion
        test('should encode links', () => {
```

Pinjasaur reviewed (approved):
LGTM. The name of the first test confused me a bit; I provided an alternative but 0/5 if you take it or not.
